lotemplate/classes.py

```diff
@@ -321,6 +391,90 @@ def fill(self, variables: dict[str, dict[str, Union[str, list[str]]]]) -> None:
         :return: None
         """
 
+        def if_replace(doc, local_variables: dict[str, dict[str, Union[str, list[str]]]]) -> None:
+            """
+            Parse statements like [if $myvar==TOTO]...[endif]
+
+            If the condition matches we remove the if and endif statement.
+            It the condition doesn't match, we remove the statements and the text between the statements.
+
+            :param doc: the document to fill
+            :param local_variables: the variables
+            :return: None
+            """
+
+            def compute_if(local_x_found):
+                """
+                Compute the if statement.
+                """
+                if_statement = IfStatement(local_x_found.getString())
+                if_result = if_statement.get_if_result(local_variables[if_statement.variable_name]['value'])
+                if not if_result:
+                    # le if n'est pas vérifié => on efface le paragraphe avec le if
+                    text = local_x_found.getText()
+                    cursor = text.createTextCursorByRange(local_x_found)
+                    cursor.goLeft(len(if_statement.if_string), False)
+                    cursor.goRight(len(if_statement.if_string), True)
+                    if not cursor.goRight(1, True):
+                        raise errors.TemplateError(
+                            'no_endif_found',
+                            f"The statement {if_statement} has no endif",
+                            dict_of(if_statement)
+                        )
+                    selected_string = cursor.String
+                    match = re.search(IfStatement.end_regex, selected_string, re.IGNORECASE)
+                    while match is None:
+                        if not cursor.goRight(1, True):
+                            raise errors.TemplateError(
+                                'no_endif_found',
+                                f"The statement {if_statement} has no endif",
+                                dict_of(if_statement)
+                            )
+                        selected_string = cursor.String
+                        match = re.search(IfStatement.end_regex, selected_string, re.IGNORECASE)
+                    cursor.String = ''
+                elif if_result:
+                    # the if is verified. We remove the statement and the endif but we keep the content
+                    position_in_text = len(if_statement.if_string)
+                    text = local_x_found.getText()
+                    cursor = text.createTextCursorByRange(local_x_found)
+                    if not cursor.goRight(1, True):
```

Comment:
This check shouldn't be in the `fill` method

Reply:
fixed